openpgp/packet/signature.go

```diff
@@ -720,7 +780,14 @@ func (sig *Signature) Serialize(w io.Writer) (err error) {
 
 func (sig *Signature) serializeBody(w io.Writer) (err error) {
 	unhashedSubpacketsLen := subpacketsLength(sig.outSubpackets, false)
-	_, err = w.Write(sig.HashSuffix[:len(sig.HashSuffix)-6])
+	fields := sig.HashSuffix[:len(sig.HashSuffix)-6]
+	if sig.Version() == 5 {
+		fields = fields[:len(fields)-4]
+		if sig.SigType == SigTypeBinary || sig.SigType == SigTypeText {
+			fields = fields[:len(fields)-6]
```

Comment:
And then also here, I would take the first `n` bytes (calculated in the same way as above) instead of everything except the last `m` bytes; or alternatively, just write the values directly (same as https://github.com/ProtonMail/crypto/blob/dbafc501ce7a096d3ff2d828c9638488c77c8cc4/openpgp/packet/signature.go#L556-L565).

Reply:
This would involve re-serializing all subpackets (via `serializeSubpackets(., ., true)`, which in turns needs a byte array of the correct length), while this is already done and stored in `HashSuffix`. I prefer to have it done once, (less operations, less errors).

Reply:
It is possible though to reparse the hashed subpacket length and still reuse the `HashSuffix` (see last commit).